pkg/extensions/chainrole/cmd.go

```diff
@@ -0,0 +1,25 @@
+package chainrole
+
+import (
+	"regexp"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	reNamespaceFilter      *regexp.Regexp
+	reServiceAccountFilter *regexp.Regexp
+)
+
+func AddCMDFlags(cmd *cobra.Command) {
+	namespacePattern := cmd.Flags().StringP("chainrole-namespace-pattern", "", "", "Namespace pattern to apply chain role functionality")
+	serviceAccountPattern := cmd.Flags().StringP("chainrole-service-account-pattern", "", "", "Service account pattern to apply chain role functionality")
+
+	if namespacePattern != nil && *namespacePattern != "" {
```

Comment:
Pretty sure `*namespacePattern != ""` is always false, since this runs in `init()` before Cobra parses the flags.

Reply:
you're absolutely right, fixed by moving regexps as part `NewCredentialsRetriever`.

Because `MustCompile` is used, this has downside of possible panic in case of invalid regexp provided from CLI, however `NewCredentialsRetriever` is created on startup, so this shouldn't be a problem.  Example of potential panic:

```go
➜  go run main.go server  --cluster-name 123123 --chainrole-namespace-pattern 123 --chainrole-service-account-pattern '\K\K' -p 9999 --probe-port 9991 --bind-hosts localhost  
2024/08/28 10:23:56 Setting logging verbosity level to: info (4)
panic: regexp: Compile(`\K\K`): error parsing regexp: invalid escape sequence: `\K`

goroutine 1 [running]:
regexp.MustCompile({0x16d8335f5, 0x4})
        /opt/homebrew/Cellar/go/1.23.0/libexec/src/regexp/regexp.go:313 +0xb0
go.amzn.com/eks/eks-pod-identity-agent/pkg/extensions/chainrole.NewCredentialsRetriever({{0x0, 0x0}, {0x102c94668, 0x14000113700}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...)
        /Users/taraspos/code/eks-pod-identity-agent/pkg/extensions/chainrole/chainrole.go:69 +0x2e8
go.amzn.com/eks/eks-pod-identity-agent/pkg/handlers.NewEksCredentialHandler({{{0x0, 0x0}, {0x102c94668, 0x14000113700}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
        /Users/taraspos/code/eks-pod-identity-agent/pkg/handlers/eks_credential_handler.go:41 +0x10c
go.amzn.com/eks/eks-pod-identity-agent/pkg/server.NewEksCredentialServer({_, _}, {{{0x0, 0x0}, {0x102c94668, 0x14000113700}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
        /Users/taraspos/code/eks-pod-identity-agent/pkg/server/server.go:62 +0x12c
go.amzn.com/eks/eks-pod-identity-agent/cmd.createServers({{0x0, 0x0}, {0x102c94668, 0x14000113700}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...})
        /Users/taraspos/code/eks-pod-identity-agent/cmd/server.go:92 +0x184
go.amzn.com/eks/eks-pod-identity-agent/cmd.startServers({_, _}, {{0x0, 0x0}, {0x102c94668, 0x14000113700}, {0x0, 0x0}, {0x0, 0x0}, ...})
        /Users/taraspos/code/eks-pod-identity-agent/cmd/server.go:66 +0x68
go.amzn.com/eks/eks-pod-identity-agent/cmd.init.func3(0x14000148200?, {0x102a7f541?, 0x4?, 0x102a7f545?})
        /Users/taraspos/code/eks-pod-identity-agent/cmd/server.go:58 +0x130
github.com/spf13/cobra.(*Command).execute(0x10302ca60, {0x140001c00c0, 0xc, 0xc})
        /Users/taraspos/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x630
github.com/spf13/cobra.(*Command).ExecuteC(0x10302c780)
        /Users/taraspos/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/taraspos/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
go.amzn.com/eks/eks-pod-identity-agent/cmd.Execute()
        /Users/taraspos/code/eks-pod-identity-agent/cmd/root.go:24 +0x24
main.main()
        /Users/taraspos/code/eks-pod-identity-agent/main.go:9 +0x1c
exit status 2
```